en/docs/apis/advanced-topics/connect-to-an-external-identity-provider.md

```diff
@@ -0,0 +1,146 @@
+    </td>
+    </tr>
+    <tr class="even">
+    <td>**Well-Known URL**</td>
+    <td><p>The OpenID Connect discovery endpoint URL. This returns the metadata related to the OpenID Provider's configuration.</p>
+    </td>
+    </tr>
+    </table>
+
+     ![Add an identity provider](../../assets/img/apis/advanced-topics/external-identity-providers/add-an-identity-provider.png){.cInlineImage-half} 
+
+     When you specify the **Well-Known URL**, the values for the **Issuer**, **Token Endpoint**, and **JWKS Endpoint** fields are filled in automatically in the next screen. Depending on your requirement, you can either keep the values as it is or manually change the values. The following table describes each field in detail:
+
+    <table>
+    <tr class="header">
+    <th><b>Field</b></th>
+    <th><b>Description</b></th>
+    </tr>
+    <tr class="odd">
+    <td>**Issuer**</td>
+    <td>The issuer identifier of the IdP, which is in the `iss` claim of the JWT issued.</td>
+    </tr>
+    <tr class="even">
+    <td>**Token Endpoint**</td>
+    <td>The token endpoint URL of the IdP from where the OAuth client can get an access token.</td>
+    </tr>
+    <tr class="odd">
+    <td>**JWKS Endpoint**</td>
+    <td>The URL that returns the JSON web key (JWK) set of the IdP. This returns a collection of JSON Web Keys used to verify the signature of the JWT tokens.
+    </td>
+    </tr>
+    </table>
+
+     ![Identity provider details](../../assets/img/apis/advanced-topics/external-identity-providers/identity-provider-details.png){.cInlineImage-half}
+
+8.  Click **Add**. This adds Okta IdP to the list of existing identity providers in your organization.
+
+Now that you have added an external IdP, you can go ahead and obtain access tokens from that IdP to exchange and get Choreo access tokens to invoke APIs.
+
+
+## Obtain an access token from an external identity provider
+
+Follow this procedure to obtain an access token from an external IdP and exchange it with a Choreo token within your organization:
+
+1. Sign in to the Choreo API Developer Portal at [https://devportal.choreo.dev/](https://devportal.choreo.dev/).
+
+2. Click the **Applications** tab.
+
+3. Create a new application or use an existing application.
+
+4. In the left navigation menu, click **OAuth2 Tokens** under **Production Keys**. This displays the following page if you have not generated the **Consumer Key** and **Consumer Secret**:
+
+    ![Generate credentials](../../assets/img/apis/advanced-topics/external-identity-providers/generate-credentials.png){.cInlineImage-full}
+
+5. Click **Generate Credentials** to generate the **Consumer Key** and **Consumer Secret**.
+
+6. Click the **User Keys** tab.
+
+    ![User keys tab](../../assets/img/apis/advanced-topics/external-identity-providers/user-keys-tab.png){.cInlineImage-half}
+   
+
+    !!! tip
+        The **User Keys** tab is only enabled if you meet the following prerequisites:</br>
```

Comment:
Shall we add these as prerequisites for this section?

Reply:
Since these requirements need to be met to enable the User Keys tab, I feel it might be confusing to add these as prerequisites related to this section. 
These are basically prerequisites to proceed with step 6.

Reply:
But then, there's no point doing anything upto step 5 if, we can't go beyond step 6 right?

Reply:
Ideally when the instructions in "Connect to an External Identity Provider"  are followed, the user would be meeting the criteria by default and can follow the steps.  Let's discuss and address in case there is any confusion on this.